File: iosApp/iosApp/Workout.swift
//
//  WorkoutModel.swift
//  iosApp
//
//  Created by Tiberiu Gradinariu on 2024-06-02.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation

struct Workout {
    var name: String
    var exercises: [Exercise]
}

struct Exercise {
    var name: String
    var sets: [Set]
}

struct Set {
    var weight: Int
    var reps: Int
}

File: shared/src/commonMain/kotlin/Workout.kt
import kotlinx.datetime.LocalDate

data class Workout(
    val name: String,
    val date: LocalDate,
    val exercises: List<Exercise>
)

data class Exercise(
    val name: String,
    val sets: List<Set>
)
enum class WeightType() {
    BODYWEIGHT,
    WEIGHTED
}

data class Set(val reps: Int, val weight: Float, val weightType: WeightType)
